### PR TITLE
New packages: phoc-0.4.2 calls-0.1.7 purism-chatty-0.1.15

### PR DESCRIPTION
**Pull request by julio641742**
Status of `phosh` on the PinePhone 1.2a

- [x] Shell
  - `phosh` `squeekboard` `phoc` `feedbackd` packaged
  - `phosh-desktop` (Should I rename it?, Does it need any additional dependencies?)
- [x] Calls (sound is bad)
  - `calls` `pinephone-wys` packaged
- [x] SMS (works) 
  -  `purism-chatty` `purple-mm-sms` packaged
- [x] Data (works)
- [ ] Camera (working on it)
  - `megapixels` packaged

The modem needs to be manually enabled every single time with `echo 1 > /sys/class/modem-power/modem-power/device/powered` (I could create a service for that).

You need to run this [script](https://gitlab.com/postmarketOS/pmaports/-/blob/master/device/community/device-pine64-pinephone/setup-modem.sh) at least one time for it to set up the modem with the right settings.

i am able to make calls with `calls` and `pinephone-wys`.  I am using the sound profiles from [here](https://gitlab.com/postmarketOS/pmaports/-/tree/master/device/community/device-pine64-pinephone/ucm) and `alsa-ucm-conf`(not yet packaged) but the sound is bad.

`purism-chatty` can't find `libjabber.so` and `mm-sms.so` that are inside `/usr/lib/purple-2`. I created symlinks from `/usr/lib/purple-2/{libjabber.so,mm-sms.so}` to `/usr/lib` Any suggestions on how to fix this?

### CODE REVIEW

File: srcpkgs/calls/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'calls'
+pkgname=calls
+version=0.1.7
+revision=1
+wrksrc=${pkgname}-v${version}
+build_style=meson
+configure_args="-Dtests=false"
```
Comment (fosslinux) [inline srcpkgs/calls/template:7]: Why? Do the tests fail, or is this just to reduce build time? We do want tests to work.

File: srcpkgs/phosh/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'phosh'
+pkgname=phosh
+version=0.4.3
+revision=1
+build_style=meson
+configure_args="-Dtests=false -Dphoc_tests=disabled"
+hostmakedepends="glib-devel gettext pkg-config wayland-devel"
+makedepends="feedbackd-devel gcr-devel gnome-desktop-devel pam-devel polkit-devel pulseaudio-devel libsecret-devel libhandy-devel NetworkManager-devel upower-devel"
```
Comment (fosslinux) [inline srcpkgs/phosh/template:8]: Split into 80 char lines, subsequent lines indented with 1 space.

**Review by fosslinux — CHANGES_REQUESTED** [review]
1. I don't understand the aggressive disabling of tests, could you please explain what the motive is here?
2. `feedbackd` should probably have two subpackages `libfeedback` and `libfeedback-devel`, keeping only the daemon in `feedbackd` itself.

**Comment by julio641742** [comment]
Ready for another review.

File: srcpkgs/feedbackd/template [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+checksum=37cf48cbdbaee8e2d0bea406f5b6317bf0e3cdf79250f9ce0a884e7d3905a8ea
+
+libfeedback_package() {
+	short_desc+=" - shared libraries"
+	pkg_install() {
+		vmove "usr/lib/*.so.*"
+	}
+}
+libfeedback-devel_package() {
+	depends="${sourcepkg}>=${version}_${revision} ${_devel_depends}"
```
Comment (fosslinux) [inline srcpkgs/feedbackd/template:25]: Whats `$_devel_depends`?

File: srcpkgs/feedbackd/template [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+homepage="https://source.puri.sm/Librem5/feedbackd"
+distfiles="${homepage}/-/archive/v${version}/${pkgname}-v${version}.tar.gz"
+checksum=37cf48cbdbaee8e2d0bea406f5b6317bf0e3cdf79250f9ce0a884e7d3905a8ea
+
+libfeedback_package() {
+	short_desc+=" - shared libraries"
+	pkg_install() {
+		vmove "usr/lib/*.so.*"
+	}
+}
```
Comment (fosslinux) [inline srcpkgs/feedbackd/template:23]: Newline after this.

**Review by fosslinux — CHANGES_REQUESTED** [review]
looks much better! :+1:

File: srcpkgs/feedbackd/template [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+libfeedback-devel_package() {
```
Comment (ericonr) [inline srcpkgs/feedbackd/template:24]: It might make sense to name this sub package `feedback-devel` instead, but I'm not sure.

**Comment by PaperMountainStudio** [comment]
I think I found some mistakes:
- phoc is missing mutter dependency
- phosh is missing at least gnome-session dependency, but then fails and asks me to log out
- /usr/bin/squeekboard-entry is not executable

thanks for the PR

File: srcpkgs/feedbackd/template (hunk 5 above)
Comment (julio641742) [reply srcpkgs/feedbackd/template:25]: Should I rename it to `feedback-libs` and `feedback-devel`?

**Comment by julio641742** [comment]
@PaperMountainStudio I did not catch these mistakes, thanks
- I added mutter to phoc
- I added the missing dependencies to phosh and it runs
- squeekboard-entry is now an executable

@fosslinux ready for another review

File: srcpkgs/feedbackd/template (hunk 5 above)
Comment (ericonr) [reply srcpkgs/feedbackd/template:25]: Honestly, given that @fosslinux suggested these specific names, I'm okay with keeping them. It was mostly an unnecessary comment.

**Comment by julio641742** [comment]
Here is phosh running on the pinephone with firefox
[img1](https://drive.google.com/file/d/1-Fz3ajXziTqBRU8owoPd9kWKeuJz0CRq/view?usp=sharing)
[img2](https://drive.google.com/file/d/1IK7FS-sGhqTsp-zhp7IqShigf8wSPBUU/view?usp=sharing)

**Comment by julio641742** [comment]
Here is a [preview](https://fosstodon.org/@julio641742/104854488643701135) of phosh on the pinephone with void
Thanks @jjsullivan5196 for packaging the kernel and to the void team for creating an amazing distro.

SMS and calls do not work :( but I am working on it.

**Comment by julio641742** [comment]
Following this [guide](https://xnux.eu/devices/feature/modem-pp.html#toc-modem-power-driver) to powerup the modem, i am able to make calls with `calls` and `pinephone-wys` with some scripts pulled from `pmaports` [setup_modem.sh](https://gitlab.com/postmarketOS/pmaports/-/blob/master/device/community/device-pine64-pinephone/setup-modem.sh) [modem.rules](https://gitlab.com/postmarketOS/pmaports/-/blob/master/device/community/device-pine64-pinephone/90-modem-eg25.rules). I am using the sound profiles from [here](https://gitlab.com/postmarketOS/pmaports/-/tree/master/device/community/device-pine64-pinephone/ucm) and `alsa-ucm-conf` but the sound is bad.

SMS works but I am not that knowleageable to make the last package [purple-mm-sms](https://source.puri.sm/Librem5/purple-mm-sms) cross compilable. To run `chatty` you need to make symlinks to point to`libjabber.so` and `mm-sms.so` that are inside `/usr/lib/purple-2`

I have not tested mobile data but it seems that `gnome-control-center` is old and does not show the Mobile tab like postmarketOs. I will try to set the `APN` settings manually later.

**Comment by jjsullivan5196** [comment]
@julio641742 I wanted to take a look packaging `alsa-ucm-conf` when I was doing the first PR, but it was already pretty fat so I didn't want to put anymore weight on it. I can look into making sure the audio config is correct for the kernel, I've messed around with the alsa config already. Also, the person maintaining the sunxi audio codecs is working on some improvements for audio quality in kernel 5.9, so there might be room to experiment with that.

Also I do have modem.rules in `pinephone-base`, but that version doesn't run the modem setup script. Seems like it would be a good idea to add that in a separate PR.

**Comment by jjsullivan5196** [comment]
@julio641742 Do you mind if I steal `atinout` for a different PR? I want to add audio routing/modem setup to pinephone-base, and it will be a crucial component.

**Comment by jjsullivan5196** [comment]
@julio641742 thanks for letting me take `atinout`, I pushed up my PR #25084 if you want to cherrypick anything :^)

**Comment by Frick-David** [comment]
Support support this!

**Comment by jcgruenhage** [comment]
Out of interest, what's missing here concretely? I'd like to give this a go and would like to avoid having to manually build updates in the future.

**Comment by jcgruenhage** [comment]
For the record, I'm working on updating the stuff in here to run it on my pinephone. `calls` has a new dependency on https://gitlab.com/mobian1/callaudiod and the associated libcallaudio, which aren't packaged yet, the rest seems to be just version bumps and libraries that are already packaged.

**Comment by jcgruenhage** [comment]
https://github.com/jcgruenhage/void-packages/tree/pinephone <- has updates for:
 - phosh and phoc
 - megapixels
 - calls (but it's broken, because my package for callaudiod doesn't split out the devel and lib packages yet)
 - squeekboard
 - mobile-config-firefox

I haven't been able to update pinephone-wys yet, since the patch doesn't apply anymore, haven't found an updated version of the patch yet.

Aside of that, after running phosh for the first time, my pinephone crashed, and now void isn't booting anymore. I think that's everything for now, but to move this forward, I'll split up the PR into multiple ones for each unrelated item, so one for calls, one for squeekboard, one for phoc and phosh, one for mobile-config-firefox and so on.

**Pull request closed without merging**